homeassistant/helpers/service.py

```diff
@@ -1159,3 +1172,47 @@ def async_register_entity_service(
         supports_response,
         job_type=job_type,
     )
+
+
+@callback
+def async_register_platform_entity_service(
+    hass: HomeAssistant,
+    component_domain: str,
+    platform_domain: str,
```

Comment:
These names are confusing I think. I think the platform domain should be media player and the component domain should be androidtv. But currently it's the other way around. The androidtv integration implements a platform under the media player domain.

`platform_name` is `androidtv` but `platform_domain` is `media_player`.

Reply:
```suggestion
    platform_domain: str,
    platform_name: str,
```

Reply:
Entity platforms, and thus entities, have two domains and we're not consistent in how we refer to them.

I don't think `platform_domain` and `platform_name` is clear either.

Just as an example, the entity registry registry calls them `domain` and `platform`, where `domain` is the IoT domain, e.g. `light` and `platform` is the domain of the integration with the platform, e.g. `hue`.

Anyway, I'll change it as you suggest to make it consistent with the entity platform helper.